### PR TITLE
Feature - Display all movie types on home screen

### PR DESCRIPTION
**Pull request by sjacobs1**
##Description
all movie arrays (Now playing, Top rated, Popular and Upcoming) display on homepage

##Screenshots
Screenshot 2024-05-05 at 4 33 22 PM

## Checklist 
Please review and check the following before submitting this pull request: 
-  The code follows the project's coding standards. 
-  All warnings Resolved.

### CODE REVIEW

File: MovieAppPetProject/MovieAppPetProject/HomeScreen/NewTableViewCell.swift [hunk 1]
```diff
@@ -10,11 +10,12 @@ import UIKit
 class NewTableViewCell: UITableViewCell, UICollectionViewDelegate, UICollectionViewDataSource {
 
     // MARK: - IBOutlets
+    @IBOutlet weak var sectionTitleLabel: UILabel!
```
Comment (ksookdhew) [inline MovieAppPetProject/MovieAppPetProject/HomeScreen/NewTableViewCell.swift]: Can this be private

**Review by giacatano — DISMISSED** [review]

File: MovieAppPetProject/MovieAppPetProject/HomeScreen/NewTableViewCell.swift (hunk 1 above)
Comment (sjacobs1) [reply MovieAppPetProject/MovieAppPetProject/HomeScreen/NewTableViewCell.swift]: yes indeed

**Review by ksookdhew — APPROVED** [review]

File: MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift [hunk 2]
```diff
@@ -14,11 +14,16 @@ class NewHomeViewController: UIViewController, UITableViewDelegate, UITableViewD
 
     // MARK: - Variables
     private lazy var nowPlayingViewModel = NowPlayingViewModel(nowPlayingRepository: NowPlayingRepository(), delegate: self)
-
+    private lazy var upcomingMoviesViewModel = UpcomingMoviesViewModel(upcomingMoviesRepository: UpcomingMoviesRepository(), delegate: self)
```
Comment (kieran-woodrow) [inline MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift:17]: so in the future, you can just have 1 viewModel that will handle the getch movie functions

File: MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift [hunk 3]
```diff
@@ -34,21 +39,55 @@ class NewHomeViewController: UIViewController, UITableViewDelegate, UITableViewD
...
-            guard let self = self else { return }
-            self.performSegue(withIdentifier: Constants.Identifiers.goToMovieDetails, sender: movie)
-        }
-        return cell
     }
 }
 
 // MARK: - Delegate
-extension NewHomeViewController: NowPlayingViewModelDelegate {
+extension NewHomeViewController: NowPlayingViewModelDelegate, UpcomingMoviesViewModelDelegate, PopularMoviesViewModelDelegate, TopRatedMoviesViewModelDelegate {
```
Comment (kieran-woodrow) [inline MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift:90]: in future, this should be avoided

File: MovieAppPetProject/MovieAppPetProject/HomeScreen/NewTableViewCell.swift [hunk 4]
```diff
@@ -10,11 +10,12 @@ import UIKit
 class NewTableViewCell: UITableViewCell, UICollectionViewDelegate, UICollectionViewDataSource {
 
     // MARK: - IBOutlets
+    @IBOutlet private weak var sectionTitleLabel: UILabel!
     @IBOutlet private weak var newCollectionView: UICollectionView!
 
     // MARK: - Variables
-    var didSelectItem: ((NowPlayingResults) -> Void)?
-    private var nowPlayingMovies: [NowPlayingResults] = []
+    var didSelectItem: ((MovieData) -> Void)?
```
Comment (kieran-woodrow) [inline MovieAppPetProject/MovieAppPetProject/HomeScreen/NewTableViewCell.swift:17]: nice protocol conformnce though

**Review by kieran-woodrow — APPROVED** [review]

File: MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift (hunk 2 above)
Comment (sjacobs1) [reply MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift:17]: understood

File: MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift (hunk 3 above)
Comment (sjacobs1) [reply MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift:90]: understood

File: MovieAppPetProject/MovieAppPetProject/HomeScreen/NewTableViewCell.swift (hunk 4 above)
Comment (sjacobs1) [reply MovieAppPetProject/MovieAppPetProject/HomeScreen/NewTableViewCell.swift:17]: thank you for the assistance !

**Pull request merged**